README.md

```diff
@@ -2,6 +2,46 @@
 
 This repository contains test suites for building proposals for CompoundGovernor / OpenZeppelin GovernorBravo. The repository is designed with [foundry](https://book.getfoundry.sh).
 
+### Example NounsDAO 500 `wstETH` -> `rETH`
+
+Want to demonstrate setting up a vote and executing a swap of 500 `wstETH` to `rETH`
+
+1. Approve the required amount of `stETH` to wrap into `wstETH`.
+2. Wrap `stETH` into `wstETH`
+3. Approve the to-be-created `Safe` (in pt. 4) to use the DAO's `wstETH`
+4. Create the `ComposableCoW` compatible `Safe` contract with:
+    - `threshold` of 1
+    - `owner` set to the `NounsDAOExecutor` address (ie. all executions on the `Safe` are bound by timelock)
+5. Execute post-Safe creation configuration and create conditional order:
+a. Set `ComposableCoW` as the domain verifier for `GPv2Settlement`
+b. Set an allowance for `GPv2VaultRelayer` to use the `wstETH` from the `Safe` contract
+c. Do `transferFrom` of the wstETH to the `Safe` contract
+d. Create the TWAP order on `ComposableCoW` via the `Safe` contract
+    - `sellToken` set to the wstETH address
+    - `buyToken` set to the rETH address
+    - `sellAmount` set to 500 wstETH
+    - `buyAmount` TBD
+    - `receiver` set to the `NounsDAOExecutor` address (all funds on swap move to the timelock)
+6. Enforce that the allowance for `wstETH` to be spent from the timelock controller is set back to zero
+
+#### Risk Mitigation
+
+1. Any `Safe` that is created that holds `NounsDAO` funds, must be subject to the timelock controller nature of `NounsDAOExecutor`.
+2. The sole owner of the `Safe` shall be the `NounsDAOExecutor`.
+3. **NO** funds are sent to the to-be-created `Safe` in the atomic transaction bundle, and are instead *pulled* from the `NounsDAOExecutor`. This avoids the unlikely situation where a pre-calculated `Safe` address (prior to `SafeProxyFactory.createProxyWithNonce`) has been determined incorrectly and funds are sent to an unrecoverable address.
+4. Approvals for spending on `NounsDAOExecutor` from the to-be-created `Safe` are zeroed at the end of the transaction bundle (though, should already be zero). 
+
+#### Verification Process
+
+1. Prove that the process can be undertaken from the context of the `NounsDAOExecutor` (ie. the timelock).
+   This means we use `vm.prank` to impersonate the `NounsDAOExecutor` and execute the swap.
+2. Do a full-stack test that:
+   - Proposes the transaction bundle
+   - Vote on the proposal (to affirm).
+   - Queue the proposal (on successful vote).
+   - Execute the proposal (verify transaction bundle executes correctly).
+   - **To be completed:** Verify via settlement that the discrete orders settle correctly.
```

Comment:
Do you already have an idea how we can achieve this? Can we "simualte" large internal balances inside the settlement contract and execute a settlement that is served from internal buffers, or do you have another idea?

Reply:
I already have in `ComposableCoW` a test simulation for TWAP, that iterates through blocks, fakes being a solver then automatically produces an opposing `GPv2Trade` for the settlement. For the opposing trade, just a simple `vm.deal` for the total `minBuyAmount`.

Then we just do sanity checks on the safe's `sellToken` balance, and the receiver's `buyToken` balance.

Reply:
This is tracked in #2